src/build/mod.rs

```diff
@@ -125,21 +129,84 @@ impl BuildContext {
     }
 
     /// Build a package from the given source specification.
+    #[instrument(skip_all, fields(source = %source_spec.source))]
     pub async fn build_source_record(
         &self,
         source_spec: &SourceRecord,
         channels: &[Url],
         target_platform: Platform,
-    ) -> Result<CondaBuildOutput, BuildError> {
-        let source = self.fetch_pinned_source(&source_spec.source).await?;
+    ) -> Result<RepoDataRecord, BuildError> {
+        let source_checkout = SourceCheckout {
+            path: self.fetch_pinned_source(&source_spec.source).await?,
+            pinned: source_spec.source.clone(),
+        };
 
-        // TODO: Add caching of this information based on the source.
+        let (cached_build, entry) = self
+            .build_cache
+            .entry(
+                &source_checkout,
+                &BuildInput {
+                    channel_urls: channels.to_vec(),
+                    target_platform: Platform::from_str(&source_spec.package_record.subdir)
+                        .ok()
+                        .unwrap_or(target_platform),
+                    name: source_spec.package_record.name.as_normalized().to_string(),
+                    version: source_spec.package_record.version.to_string(),
+                    build: source_spec.package_record.build.clone(),
+                },
+            )
+            .await?;
+
+        if let Some(build) = cached_build {
```

Comment:
Can we group the logging for debugging purposes, you can then see in one glance what was rebuild and why. I noticed this already being annoying when waiting of source packages that take long to build. This way the user can cancel the process before the builds. Can also be a future improvement in a PR.

Reply:
Decided to do this in a future PR.